EIPS/eip-5027.md

```diff
@@ -0,0 +1,84 @@
+---
+eip: 5027
+title: Unlimit contract code size
+author: Qi Zhou @qizhou
+discussions-to: https://ethereum-magicians.org/t/eip-5027-unlimit-contract-code-size/9010
+status: Draft
+type: Standards Track
+category: Core
+created: 2022-04-21
+requires: EIP-170
+---
+
+
+## Abstract
+
+Unlimit the contract code size so that users can deploy a large-code contract without worrying about splitting the contract into several sub-contracts.
+
+
+## Motivation
+
+The motivation is to
+- unlimit the code size, i.e., only limited by block gas limit;
+- minimize the changes to existing code; and
+- address the possible attack vectors in multiple contract-code-related opcodes including
+
+`CODESIZE/CODECOPY/EXTCODESIZE/EXTCODECOPY/EXTCODEHASH/DELEGATECALL/CALL/CALLCODE/STATICCALL/CREATE/CREATE2`.
+
+## Specification
+
+### Parameters
+
+| Constant                  | Value            |
+| ------------------------- | ---------------- |
+| `FORK_BLKNUM`             | TBD              |
+| `CODE_SIZE_UNIT`          | 24K              |
+| `READ_GAS_PER_UNIT`       | 700              |
+| `CREATE_DATA_GAS`         | 200              |
+
+If `block.number >= FORK_BLKNUM`, the contract creation initialization can return data with any length, but the contract-related opcodes will take extra gas as defined below:
+
+- For `CODESIZE/CODECOPY/EXTCODESIZE/EXTCODEHASH`, the gas is unchanged.
+
+- For `EXTCODECOPY/CALL/CALLCODE/DELEGATECALL/STATICCALL`, if the contract code size > `CODE_SIZE_UNIT`, then the opcodes will take extra gas as
+
+```
+(CODE_SIZE - 1) // CODE_SIZE_UNIT * READ_GAS_PER_UNIT
+```
+
+where `//` is the integer divide operator.
+
+- For CREATE/CREATE2, if the newly created contract size > `CODE_SIZE_UNIT`, the opcodes will take extra write gas as
+
+`(CODE_SIZE - CODE_SIZE_UNIT) * CREATE_DATA_GAS`.
+
+## Rationale
```

Comment:
Again, I think you slightly misunderstand the purpose of the rationale // motivation // abstract. The rationale is to explain the reasoning behind certain design decisions (e.g. why the extra gas formula was chosen).

Reply:
Thanks for the comment.  I have moved this part to the Motivation section.  Please take a look.